code/userprofile/build.gradle.kts

```diff
@@ -0,0 +1,358 @@
+            artifact(tasks.getByName<Jar>("javadocPublish"))
+
+            pom {
+                name.set(buildUtils.mavenRepoName)
+                description.set(buildUtils.mavenRepoDescription)
+                url.set(BuildUtils.DEVELOPER_DOC_URL)
+
+                licenses {
+                    license {
+                        name.set(BuildUtils.LICENSE_NAME)
+                        url.set(BuildUtils.LICENSE_URL)
+                        distribution.set(BuildUtils.LICENSE_DIST)
+                    }
+                }
+
+                developers {
+                    developer {
+                        id.set(BuildUtils.DEVELOPER_ID)
+                        name.set(BuildUtils.DEVELOPER_NAME)
+                        email.set(BuildUtils.DEVELOPER_EMAIL)
+                    }
+                }
+
+                scm {
+                    connection.set(BuildUtils.SCM_CONNECTION_URL)
+                    developerConnection.set(BuildUtils.SCM_CONNECTION_URL)
+                    url.set(BuildUtils.SCM_REPO_URL)
+                }
+
+                withXml {
+                    val dependenciesNode = asNode().appendNode("dependencies")
+
+                    val coreDependencyNode = dependenciesNode.appendNode("dependency")
+                    coreDependencyNode.appendNode("groupId", "com.adobe.marketing.mobile")
+                    coreDependencyNode.appendNode("artifactId", "core")
+                    coreDependencyNode.appendNode("version", buildUtils.mavenCoreVersion)
+                }
+            }
+        }
+    }
+
+    repositories {
+        maven {
+            name = "sonatype"
+            url = uri(buildUtils.getPublishUrl())
+            credentials {
+                username = System.getenv("SONATYPE_USERNAME")
+                password = System.getenv("SONATYPE_PASSWORD")
+            }
+        }
+    }
+}
+
+signing {
+    useGpgCmd()
+
+    setRequired(
+        tasks.withType<PublishToMavenRepository>().find {
+            gradle.taskGraph.hasTask(it)
+        }
+    )
+
+    sign(publishing.publications)
+}
+
+class BuildUtils {
+    companion object {
+        const val SNAPSHOTS_URL = "https://oss.sonatype.org/content/repositories/snapshots/"
+        const val RELEASES_URL = "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+
+        const val RELEASE_PROPERTY = "release"
+        const val JITPACK_PROPERTY = "jitpack"
+        const val SNAPSHOT_SUFFIX = "SNAPSHOT"
+        const val MAIN_SOURCE_SET = "main"
+        const val PHONE_SOURCE_SET = "phone"
+        const val UNIT_TEST_EXECUTION_RESULTS_REGEX =
+            "outputs/unit_test_code_coverage/phoneDebugUnitTest/*.exec"
+        const val FUNCTIONAL_TEST_EXECUTION_RESULTS_REGEX =
+            "outputs/code_coverage/phoneDebugAndroidTest/connected/*coverage.ec"
+
+        const val LICENSE_NAME = "The Apache License, Version 2.0"
+        const val LICENSE_URL = "http://www.apache.org/licenses/LICENSE-2.0.txt"
```

Comment:
Let's standardize on using "https" for the license file.

Reply:
Done! Updated this to be `https`